Task3/Queue/QueueExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Queue
+{
+    public static class QueueExtensions
+    {
+        public static IQueue<T> Tail<T>(this IQueue<T> queue) where T : struct
+        {
+            var tempQueue = queue;
+            var newQueue = new Queue<T>();
+            bool isFirst = true;
+
+            if (queue.IsEmpty())
+            {
+                throw new InvalidOperationException("Queue is empty!");
+            }
+
+            while (!tempQueue.IsEmpty())
+            {
+                T item = tempQueue.Dequeue();
+                if (!isFirst)
+                    newQueue.Enqueue(item);
```

Comment:
I really do not need it. Since I need to include everything except the first  item I can just store it in a temp queue  and then iterate the original queue. That way I firstly dequeue the first element an enqueue it in the tempQueue and then I can just iterate through the original queue since the first element is not already in it. That way I save more unnecessary conditions